jwst/assign_wcs/nirspec.py

```diff
@@ -368,6 +369,9 @@ def get_open_slits(input_model, reference_files=None, slit_y_range=[-.55, .55]):
         slits = [Slit('S1600A1', 3, 0, 0, 0, slit_y_range[0], slit_y_range[1], 5, 4)]
     elif exp_type == "nrs_lamp":
         slits = get_open_fixed_slits(input_model, slit_y_range)
+    elif exp_type == "nrs_autowave":
+        if lamp_mode == 'fixedslit':
```

Comment:
Should this also check for `lamp_mode=="brightobj"`?

Reply:
BTW, what are the possible values of `lamp_mode`? It may be good to add them as `enum` to the core schema.

Reply:
I'm trying to do these modes one at a time so that progress can be made rather than bundling up all the changes necessary, so I'll get to the other modes later.  I'll add an enum entry to the datamodel.

Reply:
I only mentioned `BRIGHTOBJ` because it's essentially identical to fixed slits and is already covered below in `autowave`. But I agree one at a time is best.